<!DOCTYPE html>
<html lang="en"><head>
<script src="LDA-for-microbiota_files/libs/clipboard/clipboard.min.js"></script>
<script src="LDA-for-microbiota_files/libs/quarto-html/tabby.min.js"></script>
<script src="LDA-for-microbiota_files/libs/quarto-html/popper.min.js"></script>
<script src="LDA-for-microbiota_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="LDA-for-microbiota_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="LDA-for-microbiota_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="LDA-for-microbiota_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="LDA-for-microbiota_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<link href="LDA-for-microbiota_files/libs/quarto-contrib/reveal-auto-agenda-0.0.3/reveal-auto-agenda.css" rel="stylesheet"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.450">

  <meta name="author" content="Laura Symul">
  <meta name="dcterms.date" content="2024-12-17">
  <title>Identifying microbiota sub-communities using topic models</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="LDA-for-microbiota_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="LDA-for-microbiota_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="LDA-for-microbiota_files/libs/revealjs/dist/theme/quarto.css">
  <link rel="stylesheet" href="style.css">
  <link href="LDA-for-microbiota_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="LDA-for-microbiota_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" data-background-image="images/cover.png" data-background-opacity="0.25" data-background-size="cover" class="quarto-title-block center">
  <h1 class="title">Identifying microbiota sub-communities using topic models</h1>
  <p class="subtitle">Microbiome Data Analysis Workshop 2025</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Laura Symul <a href="https://orcid.org/0000-0001-9286-0590" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a>
</div>
<div class="quarto-title-author-email">
<a href="mailto:laura.symul@uclouvain.be">laura.symul@uclouvain.be</a>
</div>
        <p class="quarto-title-affiliation">
            UCLouvain, LIDAM, ISBA
          </p>
    </div>
</div>

  <p class="date">2024-12-17</p>
</section>
<section>
<section id="hello" class="title-slide slide level1 agenda-slide center" data-background-color="#DDF1F4">
<h1>Hello üëãüëã</h1>
<div class="agenda-heading">
<p>Outline</p>
</div>
<div class="agenda">
<div class="agenda-active">
<p>Hello üëãüëã</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Introduction</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Topic models</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Choosing <span class="math inline">\(K\)</span>, the number of topics</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Conclusions &amp; references</p>
</div>
</div>
</section>
<section id="hello-1" class="slide level3">
<h3>Hello üëãüëã</h3>
<p>I‚Äôm Laura</p>
<div class="fragment">
<p><span class="emoji" data-emoji="mortar_board">üéì</span> I am an assistant professor (<span class="emoji" data-emoji="baby">üë∂</span>) in ‚Äúnon-clinical biostatistics‚Äù at UCLouvain</p>
<p><span class="emoji" data-emoji="microscope">üî¨</span> My <em>interdisciplinary</em> research agenda revolves around ‚ÄúStatistics for Women‚Äôs Health‚Äù</p>
<p><span class="emoji" data-emoji="handshake">ü§ù</span> I am part of the Vaginal Microbiome Research Consortium (<a href="https://vmrc4health.org">VMRC</a>)</p>
<p><span class="emoji" data-emoji="disguised_face">ü•∏</span> Controversy: The microbiome is <em>NOT</em> compositional in essence!</p>
<p><span class="emoji" data-emoji="climbing">üßó</span> Besides work, I love climbing (and talking (too much) about it)</p>
<p><img data-src="images/blank.png" height="20"></p>
<h4 id="code-and-data-availability">Code and data availability</h4>
<p>These slides are made with Quarto, and all analyses presented today are executed within the slides.</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-1_67d0754e368661668c7065ecb90c1a89">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># Most code is hidden by default; sometimes it can be revealed in the html as here.</span></span>
<span id="cb1-2"><a href="#cb1-2"></a></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(janitor, <span class="at">include.only =</span> <span class="st">"clean_names"</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(tidySummarizedExperiment)</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">library</span>(microViz)</span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fu">library</span>(mia)</span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="fu">library</span>(curatedMetagenomicData)</span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="fu">library</span>(topicmodels)</span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="fu">library</span>(alto) <span class="co"># devtools::install_github("lasy/alto")</span></span>
<span id="cb1-15"><a href="#cb1-15"></a></span>
<span id="cb1-16"><a href="#cb1-16"></a>tmp <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_map</span>(<span class="st">"R/"</span>, source)</span>
<span id="cb1-17"><a href="#cb1-17"></a></span>
<span id="cb1-18"><a href="#cb1-18"></a><span class="fu">theme_set</span>(<span class="fu">theme_light</span>())</span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>You can find the source for this slidedeck on GitHub: <a href="github.com/lasy/t-2425-Microbiota-UHasselt">lasy/t-2425-Microbiota-UHasselt</a></p>
<p>Data used here are publicly available and linked in the slides.</p>
<h4 id="feedback">Feedback</h4>
<p><span class="emoji" data-emoji="pray">üôè</span> Please give me feedback or report mistakes/typos (at the end of the workshop in person, or on GitHub using <em>Issues</em>)</p>
<h4 id="cc-by-sa">CC-BY-SA</h4>
</div>
</section></section>
<section>
<section id="introduction" class="title-slide slide level1 agenda-slide center" data-background-color="#DDF1F4">
<h1>Introduction</h1>
<div class="agenda-heading">
<p>Outline</p>
</div>
<div class="agenda">
<div class="agenda-inactive agenda-pre-active">
<p>Hello üëãüëã</p>
</div>
<div class="agenda-active">
<p>Introduction</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Topic models</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Choosing <span class="math inline">\(K\)</span>, the number of topics</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Conclusions &amp; references</p>
</div>
</div>
</section>
<section id="running-examples" class="title-slide slide level2 center">
<h2>Running Examples</h2>
<p>We‚Äôll use two microbiota datasets:</p>
<ul>
<li><p>One vaginal microbiota dataset: The ISALA dataset <span class="citation" data-cites="isala">(<a href="#/references" role="doc-biblioref" onclick="">Lebeer et al. 2023</a>)</span> <img data-src="images/logo-Isala.png" width="30"></p></li>
<li><p>One gut microbiota dataset (LLD study) from the <code>curatedMetagenomicsData</code> Bioconductor package <span class="citation" data-cites="curatedMg">(<a href="#/references" role="doc-biblioref" onclick="">Pasolli et al. 2017</a>)</span></p></li>
</ul>
</section>
<section id="vaginal-microbiota-dataset-isala" class="slide level3">
<h3>Vaginal microbiota dataset: ISALA</h3>
<div class="columns">
<div class="column" style="width:30%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="images/logo-Isala.png" width="200"></p>
</figure>
</div>
</div><div class="column" style="width:70%;">
<p><a href="https://isala.be/en/">ISALA</a> is a citizen-science project from Sarah Lebeer‚Äôs lab @ UAntwerp, Belgium.</p>
<p>The project and data have been described in a Nature Microbiology paper <span class="citation" data-cites="isala">(<a href="#/references" role="doc-biblioref" onclick="">Lebeer et al. 2023</a>)</span>.</p>
<p>De-identified data for over 3000 samples is available on the <a href="https://github.com/LebeerLab/Citizen-science-map-of-the-vaginal-microbiome/tree/main">LebeerLab GitHub</a> repository as <code>tsv</code> files.</p>
</div>
</div>
<div class="fragment">
<p>We‚Äôll use the data aggregated at the Genus level (but Species level for <em>Lactobacillus</em> spp.) for this example.</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-2_275c191f1dfeaeb8b59bd0618015423f">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>github_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/LebeerLab/Citizen-science-map-of-the-vaginal-microbiome/refs/heads/main/data/isala/"</span></span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a>isala_counts_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(github_url, <span class="st">"counts.tsv"</span>))</span>
<span id="cb2-4"><a href="#cb2-4"></a>isala_tax_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(github_url, <span class="st">"taxa_spec.tsv"</span>))</span>
<span id="cb2-5"><a href="#cb2-5"></a>isala_metadata_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="fu">str_c</span>(github_url, <span class="st">"metadata.tsv"</span>))</span>
<span id="cb2-6"><a href="#cb2-6"></a></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="fu">rm</span>(github_url)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-4_c9a12bed4bce52739805ca56ec3eb6cd">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>isala_full <span class="ot">&lt;-</span> </span>
<span id="cb3-2"><a href="#cb3-2"></a>  <span class="fu">make_se_from_isala_raw</span>(</span>
<span id="cb3-3"><a href="#cb3-3"></a>    isala_counts_raw, </span>
<span id="cb3-4"><a href="#cb3-4"></a>    isala_tax_raw, </span>
<span id="cb3-5"><a href="#cb3-5"></a>    isala_metadata_raw</span>
<span id="cb3-6"><a href="#cb3-6"></a>    )</span>
<span id="cb3-7"><a href="#cb3-7"></a></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="fu">rm</span>(isala_counts_raw, isala_tax_raw, isala_metadata_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-5_a1ee826a0cf3e73e9a6e0a4dceec62f9">
<div class="cell-output cell-output-stdout">
<pre><code># A SummarizedExperiment-tibble abstraction: 1,661,478 √ó 38
# Features=546 | Samples=3043 | Assays=counts, rel_ab
   .feature           .sample  counts  rel_ab general_age health_bmi health_antibiotic_3m‚Ä¶¬π sexual_intercourse_2‚Ä¶¬≤ technical_run_202010‚Ä¶¬≥ technical_run_202010‚Ä¶‚Å¥
   &lt;chr&gt;              &lt;chr&gt;     &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt;      &lt;dbl&gt; &lt;lgl&gt;                  &lt;lgl&gt;                  &lt;lgl&gt;                  &lt;lgl&gt;                 
 1 Methanobrevibacter SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 2 Methanosphaera     SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 3 CP012850           SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 4 JF807641           SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 5 HQ597814           SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 6 GU260705           SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 7 LCZK               SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 8 Actinobaculum      SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
 9 Actinomyces        SAMEA13‚Ä¶    170 0.00433          29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
10 Actinotignum       SAMEA13‚Ä¶      0 0                29       18.9 TRUE                   FALSE                  TRUE                   FALSE                 
# ‚Ñπ 40 more rows
# ‚Ñπ abbreviated names: ¬π‚Äãhealth_antibiotic_3months, ¬≤‚Äãsexual_intercourse_24hours, ¬≥‚Äãtechnical_run_20201009_isala_cross_1, ‚Å¥‚Äãtechnical_run_20201016_isala_cross_2
# ‚Ñπ 28 more variables: technical_run_20201023_isala_cross_3 &lt;lgl&gt;, technical_run_20201030_isala_cross_4 &lt;lgl&gt;, technical_run_20201106_isala_cross_5 &lt;lgl&gt;,
#   technical_run_20201120_isala_cross_6 &lt;lgl&gt;, technical_run_20201204_isala_cross_8 &lt;lgl&gt;, technical_run_20201211_isala_cross_9 &lt;lgl&gt;,
#   technical_run_20210119_isala_cross_7 &lt;lgl&gt;, technical_libsize &lt;dbl&gt;, plate &lt;chr&gt;, volume &lt;dbl&gt;, dna_conc &lt;dbl&gt;, read_conc &lt;dbl&gt;, read_conc_norm &lt;dbl&gt;,
#   lib_size &lt;dbl&gt;, sample_month &lt;dbl&gt;, transit_time &lt;dbl&gt;, valencia_subcst &lt;chr&gt;, valencia_cst &lt;chr&gt;, max_genus1 &lt;chr&gt;, max_genus2 &lt;chr&gt;, kingdom &lt;chr&gt;,
#   phylum &lt;chr&gt;, class &lt;chr&gt;, order &lt;chr&gt;, family &lt;chr&gt;, genus &lt;chr&gt;, taxon_id &lt;chr&gt;, last_available_level &lt;chr&gt;</code></pre>
</div>
</div>
</div>
</section>
<section id="vaginal-microbiota-dataset-isala-1" class="slide level3">
<h3>Vaginal microbiota dataset: ISALA</h3>
<p>We take a random subset of 200 samples (out of ) to make it easier to work with.</p>
<div class="fragment">
<p>We then also filter out ‚Äúrare taxa‚Äù:</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-10_461f04b342837ec798a44194420ada9d">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-10-1.png" width="576"></p>
</div>
</div>
<p>This reduces the number of taxa from 546 to 60.</p>
</div>
</section>
<section id="gut-microbiota-dataset-curatedmetagenomics" class="slide level3">
<h3>Gut microbiota dataset: <code>curatedMetagenomics</code></h3>
<p>We‚Äôll use data from the <a href="https://bioconductor.org/packages/release/data/experiment/html/curatedMetagenomicData.html"><code>curatedMetagenomicData</code> Bioconductor package</a>.</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="images/curatedMetagenomicData.png" width="1200"></p>
</figure>
</div>
</section>
<section id="gut-microbiota-dataset-curatedmetagenomics-1" class="slide level3">
<h3>Gut microbiota dataset: <code>curatedMetagenomics</code></h3>
<p>and more specifically, data from the <code>LifeLinesDeep 2016</code> study.</p>
<div class="fragment">
<p>As for the ISALA data, we create a <code>SummarizedExperiment</code> object from the package data.</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-13_187faf1a0a76e4a240cde55cd6469179">
<div class="cell-output cell-output-stdout">
<pre><code># A SummarizedExperiment-tibble abstraction: 722,995 √ó 32
# Features=637 | Samples=1135 | Assays=counts, rel_ab
   .feature              .sample counts  rel_ab study_name subject_id body_site antibiotics_current_‚Ä¶¬π study_condition disease   age age_category gender country
   &lt;chr&gt;                 &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;                  &lt;chr&gt;           &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;  &lt;chr&gt;  
 1 [Bacteroides] pectin‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 2 [Butyribacterium] me‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 3 [Clostridium] hylemo‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 4 [Clostridium] innocu‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 5 [Clostridium] leptum  EGAR00‚Ä¶   9031 2.88e-4 LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 6 [Clostridium] methyl‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 7 [Clostridium] scinde‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 8 [Clostridium] spirof‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
 9 [Clostridium] symbio‚Ä¶ EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
10 [Eubacterium] brachy  EGAR00‚Ä¶      0 0       LifeLines‚Ä¶ sub_90020‚Ä¶ stool     no                     control         healthy    35 adult        female NLD    
# ‚Ñπ 40 more rows
# ‚Ñπ abbreviated name: ¬π‚Äãantibiotics_current_use
# ‚Ñπ 18 more variables: non_westernized &lt;chr&gt;, sequencing_platform &lt;chr&gt;, DNA_extraction_kit &lt;chr&gt;, PMID &lt;chr&gt;, number_reads &lt;int&gt;, number_bases &lt;dbl&gt;,
#   minimum_read_length &lt;int&gt;, median_read_length &lt;int&gt;, curator &lt;chr&gt;, BMI &lt;dbl&gt;, population &lt;chr&gt;, superkingdom &lt;chr&gt;, phylum &lt;chr&gt;, class &lt;chr&gt;,
#   order &lt;chr&gt;, family &lt;chr&gt;, genus &lt;chr&gt;, species &lt;chr&gt;</code></pre>
</div>
</div>
</div>
</section>
<section id="gut-microbiota-dataset-curatedmetagenomics-2" class="slide level3">
<h3>Gut microbiota dataset: <code>curatedMetagenomics</code></h3>
<p>As for the ISALA data, we take a random subset of 200 samples (out of 1135) to make it easier to work with.</p>
<div class="fragment">
<p>We then also filter out ‚Äúrare taxa‚Äù:</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-16_2c91e3f8007aede7d87f7fa26a93e716">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-16-1.png" width="576"></p>
</div>
</div>
<p>This reduces the number of taxa from 637 to 216.</p>
</div>
</section>

<section id="quick-exploration-and-comparison-of-the-data-microbiota-diversity" class="title-slide slide level2 center">
<h2>Quick exploration and comparison of the data: Microbiota diversity</h2>

</section>
<section id="alpha-diversity" class="slide level3">
<h3><span class="math inline">\(\alpha\)</span>-diversity</h3>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-21_4f39d4aff02dded9af51c7902ece2eed">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-21-1.png" width="960"></p>
</div>
</div>
<p><span class="emoji" data-emoji="arrow_right">‚û°Ô∏è</span> Vaginal swabs <span class="math inline">\(\alpha\)</span>-diversity is lower than stool samples <span class="math inline">\(\alpha\)</span>-diversity.</p>
</section>
<section id="beta-diversity" class="slide level3">
<h3><span class="math inline">\(\beta\)</span>-diversity</h3>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-24_10fc9410643d439294659cebdf1ae1f6">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-24-1.png" width="960"></p>
</div>
</div>
<p><span class="emoji" data-emoji="arrow_right">‚û°Ô∏è</span> Vaginal swabs <span class="math inline">\(\beta\)</span>-diversity has a wide distribution: some samples are very similar (complete overlap), others are very different (no overlap).</p>
</section>

<section id="any-typical-groups-or-types" class="title-slide slide level2 center">
<h2>Any typical groups or types?</h2>

</section>
<section id="any-typical-microbiota-groups-or-types" class="slide level3">
<h3>Any typical microbiota groups or types?</h3>
<div class="columns">
<div class="column" style="width:50%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-26_ffe9c871c0851df8189c2799fc7b7860">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-26-1.png" width="576"></p>
</div>
</div>
<p>ISALA (vaginal swabs)</p>
</div><div class="column fragment" style="width:50%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-27_94a04818814a1edb3fa52270c0560448">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-27-1.png" width="576"></p>
</div>
</div>
<p>LLD (stool) - aggregated at the Family level</p>
</div>
</div>
</section>

<section id="subcommunities-in-microbiotas" class="title-slide slide level2 center">
<h2>‚ÄúSubcommunities‚Äù in microbiotas</h2>
<p><span class="emoji" data-emoji="warning">‚ö†Ô∏è</span> Note: bacterial ‚Äú<em>subcommunity</em>‚Äù is not a well defined term in microbiology.</p>
<p>But it aims to reflect the consequences of well establish concepts (<em>i.e.</em>, bacterial interactions).</p>
</section>
<section id="bacterial-interactions" class="slide level3">
<h3>Bacterial interactions</h3>
<p>Many bacteria interact with each other in many ways, mostly through metabolic exchanges.</p>
<div class="fragment">
<div class="columns">
<div class="column fragment" style="width:50%;">
<p><strong>Negative interactions</strong></p>
<ul>
<li>Competition (for the same nutrients)</li>
</ul>
<p><img data-src="images/interactions_competition.png" width="200"></p>
<ul>
<li>Direct inhibition (<em>e.g.</em>, by producing compounds toxic to other bacteria)</li>
</ul>
<p><img data-src="images/interactions_inhibition.png" width="200"></p>
<p>For example, in the vaginal microbiota, <em>lactobicilli</em> produce lactic acid, which inhibits the growth of other bacteria.</p>
</div><div class="column fragment" style="width:50%;">
<p><strong>Positive interactions</strong></p>
<ul>
<li>Commensalism (<em>e.g.</em>, one species benefits from another one without affecting it)</li>
</ul>
<p><img data-src="images/interactions_commensalism.png" width="200"></p>
<ul>
<li>Mutualism (cooperation or cross-feeding (<em>e.g.</em>, bacteria feed on each other wastes)</li>
</ul>
<p><img data-src="images/interactions_mutualism.png" width="200"></p>
</div>
</div>
<aside class="notes">
<p>For species-species interactions, see: https://en.wikipedia.org/wiki/Symbiosis</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</div>
</section>
<section id="bacterial-interactions-1" class="slide level3">
<h3>Bacterial interactions</h3>
<p><img data-src="images/blank.png" height="50"></p>
<p>In microbiotas, this leads to complex networks of interactions.</p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="images/interactions_network.png" width="800"></p>
</figure>
</div>
</section>
<section id="subcommunities" class="slide level3">
<h3>Subcommunities</h3>
<p>Bacteria-bacteria interactions lead to non-random co-occurence patterns.</p>
<div class="fragment">
<p>In the context of microbiotas, <em>subcommunities</em> are groups of bacteria that frequently co-occur.</p>
</div>
<div class="fragment">
<div class="r-stack">
<p><img data-src="images/interactions_subcommunities_empty.png" class="fragment" width="450"></p>
<p><img data-src="images/interactions_subcommunities.png" class="fragment" width="450"></p>
</div>
<p><img data-src="images/blank.png" class="fragment" height="50"></p>
</div>
<div class="fragment">
<ul>
<li><p>Subcommunities can co-exist in the same microbiota</p></li>
<li><p>There can be <em>transitions</em> between subcommunity dominance</p></li>
</ul>
</div>
</section>

<section id="duality-types-subcommunities" class="title-slide slide level2 center">
<h2>Duality: ‚Äútypes‚Äù ‚áî subcommunities</h2>
<p>If there are subcommunities, we expect microbiota ‚Äútypes‚Äù to be associated with them.</p>
<div class="fragment">
<p>And if there are microbiota ‚Äútypes‚Äù, we expect subcommunities membership to match the composition of prototypical ‚Äútype‚Äù samples.</p>
</div>
</section>
</section>
<section>
<section id="topic-models" class="title-slide slide level1 agenda-slide center" data-background-color="#DDF1F4">
<h1>Topic models</h1>
<div class="agenda-heading">
<p>Outline</p>
</div>
<div class="agenda">
<div class="agenda-inactive agenda-pre-active">
<p>Hello üëãüëã</p>
</div>
<div class="agenda-inactive agenda-pre-active">
<p>Introduction</p>
</div>
<div class="agenda-active">
<p>Topic models</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Choosing <span class="math inline">\(K\)</span>, the number of topics</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Conclusions &amp; references</p>
</div>
</div>
</section>
<section id="what-are-topic-models" class="title-slide slide level2 center">
<h2>What are topic models?</h2>

</section>
<section id="topic-models-1" class="slide level3">
<h3>Topic models</h3>
<p>Topic models are statistical models for identifying ‚Äú<em>topics</em>‚Äù in corpus of text documents.</p>
<p><img data-src="images/LDA_illlustration.png" style="width:16.66667in"></p>
<div class="fragment">
<p>Several Bayesian and non-Bayesian models have been proposed:</p>
<ul>
<li><strong>Latent Dirichlet Allocation</strong> (LDA) <span class="citation" data-cites="lda1 lda">(<a href="#/references" role="doc-biblioref" onclick="">Pritchard, Stephens, and Donnelly 2000</a>; <a href="#/references" role="doc-biblioref" onclick="">Blei, Ng, and Jordan 2003</a>)</span></li>
<li>and derived models (<em>e.g.</em>, Hierarchical LDA)</li>
<li>Non-negative Matrix Factorization (NMF)</li>
<li>(probabilistic) Latent Semantic Analysis/Indexing ((p)LSA/LSI)</li>
<li>‚Ä¶</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>There are also ‚Äúsupervised‚Äù versions of topic models</li>
</ul>
</div>
</section>
<section id="lda" class="slide level3">
<h3>LDA</h3>
<div class="columns">
<div class="column" style="width:30%;">
<p><img data-src="images/LDA_plate.png" width="200"></p>
<p><span class="citation" data-cites="lda">(<a href="#/references" role="doc-biblioref" onclick="">Blei, Ng, and Jordan 2003</a>)</span></p>
</div><div class="column" style="width:70%;">
<p><img data-src="images/LDA_illlustration.png" width="700"></p>
</div>
</div>
<div class="fragment">
<p>Parameter <em>inference</em> is typically done using Gibbs sampling or variational inference.</p>
</div>
<div class="fragment">
<p>Several LDA implementations are available in <code>R</code>; among the best is the <code>topicmodels</code> package <span class="citation" data-cites="topicmodels">(<a href="#/references" role="doc-biblioref" onclick="">Gr√ºn and Hornik 2011</a>)</span>.</p>
</div>
</section>

<section id="lda-for-microbiota-data" class="title-slide slide level2 center">
<h2>LDA for microbiota data?</h2>

</section>
<section id="lda-for-microbiota-data-1" class="slide level3">
<h3>LDA for microbiota data</h3>
<div class="columns">
<div id="vcenter" class="column" style="width:60%;">
<p><img data-src="images/LDA_illlustration.png" width="700"> <img data-src="images/LDA_illlustration_microbiota.png" width="700"></p>
</div><div id="vcenter" class="column" style="width:40%;">
<table>
<thead>
<tr class="header">
<th>NLP</th>
<th>Microbiota analyses</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Documents</td>
<td>samples</td>
</tr>
<tr class="even">
<td>Words</td>
<td>taxa (ASV, species, genus, ‚Ä¶)</td>
</tr>
<tr class="odd">
<td>Topics</td>
<td>sub-communities</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="citation" data-cites="sankaran_latent_2019 subcommunities">(<a href="#/references" role="doc-biblioref" onclick="">Sankaran and Holmes 2019</a>; <a href="#/references" role="doc-biblioref" onclick="">Symul et al. 2023</a>)</span></p>
</section>
<section id="lda-on-microbiota-data" class="slide level3">
<h3>LDA on microbiota data</h3>
<div class="panel-tabset">
<ul id="tabset-1" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-1-1">ISALA (vag. swabs)</a></li><li><a href="#tabset-1-2">LifeLines DEEP (stool)</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-30_16ff23baab1934e20bd6b2ed8be300eb">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>isala_tse <span class="ot">&lt;-</span> isala <span class="sc">|&gt;</span> mia<span class="sc">::</span><span class="fu">addLDA</span>(<span class="at">k =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="columns">
<div id="vcenter" class="column" style="width:35%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-32_5434555d189802c780b5aac9ef300fb2">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-32-1.png" width="672"></p>
</div>
</div>
</div><div id="vcenter" class="column fragment" style="width:5%;">
<p>~</p>
</div><div id="vcenter" class="column fragment" style="width:20%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-33_91f5693c855b0506c0c8e47a143939e3">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-33-1.png" width="336"></p>
</div>
</div>
</div><div class="column fragment" style="width:35%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-34_687caced4b771833e237daaec867f16b">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-34-1.png" width="672"></p>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-1-2">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-35_e17b6a8843797bd62d9d3045039ca5c0">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>LLD_LDA <span class="ot">&lt;-</span> LLD_family <span class="sc">|&gt;</span>  <span class="fu">assay</span>(<span class="st">"counts"</span>) <span class="sc">|&gt;</span> <span class="fu">t</span>() <span class="sc">|&gt;</span> topicmodels<span class="sc">::</span><span class="fu">LDA</span>(<span class="at">k =</span> <span class="dv">6</span>, <span class="at">method =</span> <span class="st">"VEM"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="columns">
<div id="vcenter" class="column" style="width:35%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-37_942ac605c75b5f517d8a3c00c95ea382">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-37-1.png" width="672"></p>
</div>
</div>
</div><div id="vcenter" class="column fragment" style="width:5%;">
<p>~</p>
</div><div id="vcenter" class="column fragment" style="width:20%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-38_119325ff107c9ce232e8c33241e7827a">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-38-1.png" width="336"></p>
</div>
</div>
</div><div class="column fragment" style="width:35%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-39_1ec61b88dfe36f47220eacc45161931f">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-39-1.png" width="672"></p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</section>

<section id="lda-vs-clustering" class="title-slide slide level2 center">
<h2>LDA <em>vs</em> clustering</h2>

</section>
<section id="lda-vs-clustering-1" class="slide level3">
<h3>LDA <em>vs</em> clustering</h3>
<div class="columns">
<div class="column" style="width:50%;">
<p><img data-src="images/LDA_vs_clustering_C.png"> <img data-src="images/LDA_vs_clustering_LDA.png"></p>
</div><div class="column fragment" style="width:50%;">
<p><img data-src="images/lebeer_PCoA.png" height="500"></p>
<p>from <span class="citation" data-cites="isala">(<a href="#/references" role="doc-biblioref" onclick="">Lebeer et al. 2023</a>)</span>. Valencia is a vaginal microbiota ‚Äúclassifier‚Äù <span class="citation" data-cites="valencia">(<a href="#/references" role="doc-biblioref" onclick="">France et al. 2020</a>)</span> built from a large colated dataset.</p>
</div>
</div>
</section>
<section id="lda-vs-clustering-2" class="slide level3">
<h3>LDA <em>vs</em> clustering</h3>
<ul>
<li><p>In longitudinal data, clustering can‚Äôt capture the <em>rate</em> of transitions</p></li>
<li><p>Mixed memberships (topic models) are more powerful than categorical memberships (clustering) to find associations with responses.</p></li>
<li><p>‚ÄúTransition‚Äù clusters may group samples that are on very different trajectories.</p></li>
</ul>
</section>
</section>
<section>
<section id="choosing-k-the-number-of-topics" class="title-slide slide level1 agenda-slide center" data-background-color="#DDF1F4">
<h1>Choosing <span class="math inline">\(K\)</span>, the number of topics</h1>
<div class="agenda-heading">
<p>Outline</p>
</div>
<div class="agenda">
<div class="agenda-inactive agenda-pre-active">
<p>Hello üëãüëã</p>
</div>
<div class="agenda-inactive agenda-pre-active">
<p>Introduction</p>
</div>
<div class="agenda-inactive agenda-pre-active">
<p>Topic models</p>
</div>
<div class="agenda-active">
<p>Choosing <span class="math inline">\(K\)</span>, the number of topics</p>
</div>
<div class="agenda-inactive agenda-post-active">
<p>Conclusions &amp; references</p>
</div>
</div>
</section>
<section id="perplexity-based-methods" class="title-slide slide level2 center">
<h2>Perplexity-based methods</h2>

</section>
<section id="log-likelihood-and-perplexity" class="slide level3">
<h3>Log-likelihood and Perplexity</h3>
<p>The log-likelihood of observing counts <span class="math inline">\(\mathbf{X}\)</span> (<span class="math inline">\(I \times J\)</span>) given a model <span class="math inline">\(M\)</span> with parameters <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> is</p>
<p><span class="math display">\[\mathcal{L}(\mathbf{X}|M) = \mathcal{L}(\mathbf{X}|\beta, \gamma) = \sum_{i=1}^{I} \log p(\mathbf{x}_{i.}|\beta, \gamma) = \sum_{i = 1}^I \sum_{j = 1}^J x_{ij} \log(\beta \ \gamma)_{ij}\]</span></p>
<div class="fragment">
<p>The perplexity <span class="math inline">\(PP\)</span> is then defined as <span class="math inline">\(PP(\mathbf{X}) = \exp\left(- \frac{ \mathcal{L}(\mathbf{X}|\beta, \gamma)}{\sum_{i = 1}^I \sum_{j = 1}^J x_{ij}} \right)\)</span></p>
<p>See <span class="citation" data-cites="topicmodels">(<a href="#/references" role="doc-biblioref" onclick="">Gr√ºn and Hornik 2011</a>)</span> for details.</p>
</div>
<div class="fragment">
<p>The perplexity is expected to decrease as the number of topics increases, so we look for an elbow.</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-42_6a3ec62ae9a91dfc58f3e75b4665a0b7">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>ks <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">15</span></span>
<span id="cb8-2"><a href="#cb8-2"></a></span>
<span id="cb8-3"><a href="#cb8-3"></a>isala_tse_1 <span class="ot">&lt;-</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  isala <span class="sc">|&gt;</span> </span>
<span id="cb8-5"><a href="#cb8-5"></a>  mia<span class="sc">::</span><span class="fu">addLDA</span>(<span class="at">k =</span> ks, <span class="at">name =</span> <span class="st">"LDA_multiple_k"</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">seed =</span> <span class="dv">1</span>))</span>
<span id="cb8-6"><a href="#cb8-6"></a>isala_tse_2 <span class="ot">&lt;-</span> </span>
<span id="cb8-7"><a href="#cb8-7"></a>  isala <span class="sc">|&gt;</span> </span>
<span id="cb8-8"><a href="#cb8-8"></a>  mia<span class="sc">::</span><span class="fu">addLDA</span>(<span class="at">k =</span> ks, <span class="at">name =</span> <span class="st">"LDA_multiple_k"</span>, <span class="at">control =</span> <span class="fu">list</span>(<span class="at">seed =</span> <span class="dv">9</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-43_1c3e7139adc30968b5004443b56ec00b">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-43-1.png" width="960"></p>
</div>
</div>
<p>Note that since LDA is not deterministic, the perplexity may vary between runs.</p>
</div>
</section>
<section id="perplexity-based-cross-validation" class="slide level3">
<h3>Perplexity-based cross-validation</h3>
<p>We can also choose <span class="math inline">\(K\)</span> based on the perplexity in cross-validation.</p>
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-47_f02235ee39a7f843a1f9bcafd6019e99">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-47-1.png" width="960"></p>
</div>
</div>
</section>

<section id="alignement-across-k" class="title-slide slide level2 center">
<h2>Alignement across K</h2>
<p><span class="citation" data-cites="alto">(<a href="#/references" role="doc-biblioref" onclick="">Fukuyama, Sankaran, and Symul 2023</a>)</span></p>
</section>
<section id="general-idea" class="slide level3">
<h3>General idea</h3>
<p>If some true <span class="math inline">\(K^*\)</span> sub-communities exist, we expect the corresponding topics to be robustly identified by LDA at any <span class="math inline">\(K \geq K^*\)</span>.</p>
<div class="fragment">
<p>So, topics that are similar across different <span class="math inline">\(K\)</span> are more likely to correspond to these true sub-communities.</p>
</div>
<div class="fragment">
<p>And topics that are not similar across different <span class="math inline">\(K\)</span> are more likely to be spurious.</p>
</div>
<div class="fragment">
<p>How to identify topics that are similar across different <span class="math inline">\(K\)</span>?</p>
</div>
</section>
<section id="matching-topics-across-k-aligning-them" class="slide level3">
<h3>Matching topics across <span class="math inline">\(K\)</span> (‚ÄúAligning‚Äù them)</h3>
<p>How to match topics across different <span class="math inline">\(K\)</span>?</p>
<div class="fragment">
<p>Topics that are present in the same samples are likely to be similar.</p>
<div class="panel-tabset">
<ul id="tabset-2" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-2-1">ISALA (vag. swabs)</a></li><li><a href="#tabset-2-2">LLD (stool) - family level</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-50_022be1dfc7946e1da721d6fce3970ef0">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-50-1.png" width="960"></p>
</div>
</div>
</div>
<div id="tabset-2-2">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-51_9bb981106e32dece4268c3c2a74eadf8">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-51-1.png" width="960"></p>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="matching-topics-across-k-aligning-them-1" class="slide level3">
<h3>Matching topics across <span class="math inline">\(K\)</span> (‚ÄúAligning‚Äù them)</h3>
<p>or‚Ä¶</p>
<p>Topics that have a similar composition are likely to be similar.</p>
<div class="panel-tabset">
<ul id="tabset-3" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-3-1">ISALA (vag. swabs)</a></li><li><a href="#tabset-3-2">LLD (stool) - family level</a></li></ul>
<div class="tab-content">
<div id="tabset-3-1">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-52_3ba8563577a9f27c8ac15d1e1692a7ce">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-52-1.png" width="960"></p>
</div>
</div>
</div>
<div id="tabset-3-2">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-53_9c21f39fbb6a4ae6b9d16bd059f4a154">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-53-1.png" width="960"></p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="duality-beta-gamma" class="slide level3">
<h3>Duality: <span class="math inline">\(\beta\)</span> &amp; <span class="math inline">\(\gamma\)</span></h3>
<p>‚ÄúTopics that are present in the same samples are likely to be similar‚Äù</p>
<div class="fragment">
<p><span class="math inline">\(\rightarrow\)</span> Alignment based on the <span class="math inline">\(\gamma\)</span> matrix (proportion of each topic in each sample)</p>
</div>
<div class="fragment">
<p><img data-src="images/blank.png" height="50"></p>
<p>‚ÄúTopics that have a similar composition are likely to be similar‚Äù</p>
</div>
<div class="fragment">
<p><span class="math inline">\(\rightarrow\)</span> Alignment based on the <span class="math inline">\(\beta\)</span> matrix (probability of each taxon in each topic)</p>
</div>
</section>
<section id="sample-composition-based-alignement-gamma" class="slide level3">
<h3>Sample-composition-based alignement (<span class="math inline">\(\gamma\)</span>)</h3>
<div class="columns">
<div id="vcenter" class="column" style="width:35%;">
<p><img data-src="images/alto_product.png" height="200"></p>
</div><div id="vcenter" class="column" style="width:65%;">
<p>Alignment weights for the product method: <span class="math inline">\(w_{jk} = \sum_{i=1}^{N}\gamma_{ij}^{(1)} \gamma_{ik}^{(2)}\)</span></p>
<p>where <span class="math inline">\(\gamma_{ij}^{(1)}\)</span> are the proportions of topic <span class="math inline">\(j\)</span> in sample <span class="math inline">\(i\)</span> for model 1.</p>
</div>
</div>
<div class="fragment">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-55_5d4ab33d214456ac18481c71277ccc85">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-55-1.png" width="864"></p>
</div>
</div>
</div>
</section>
<section id="topic-composition-based-alignement-beta" class="slide level3">
<h3>Topic-composition-based alignement (<span class="math inline">\(\beta\)</span>)</h3>
<div class="columns">
<div id="vcenter" class="column" style="width:35%;">
<p><img data-src="images/alto_transport.png" height="200"></p>
</div><div id="vcenter" class="column" style="width:65%;">
<p>Alignment weights are obtained by solving the optimal transport problem</p>
<p><span class="math display">\[\min_{W \ \in \ \mathcal{U}(p, q)} \langle C, W\rangle\]</span></p>
<p><span class="math display">\[\mathcal{U}(p,q) := \{W \in R^{|V_p|\times|V_q|}_+ : W \mathbf{1}_{|V_q|} = p \text{ and }  W^{T} \mathbf{1}_{|V_p|} = q\}\]</span></p>
<p>where <span class="math inline">\(C(v, v') := JSD(\beta(v), \beta(v'))\)</span> is the <strong>cost</strong> of transporting mass between topics <span class="math inline">\(v\)</span> and <span class="math inline">\(v'\)</span>.</p>
</div>
</div>
<div class="fragment">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-57_7cd489b38b30c97868441e9d316d81e6">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-57-1.png" width="864"></p>
</div>
</div>
</div>
</section>
<section id="alignment-plots" class="slide level3">
<h3>Alignment plots</h3>
<p><img data-src="images/alto_illustration.png" height="400"></p>
</section>
<section id="alignment-plots-1" class="slide level3">
<h3>Alignment plots</h3>
<div class="panel-tabset">
<ul id="tabset-4" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-4-1">Isala (vag. swabs)</a></li><li><a href="#tabset-4-2">LLD (stool)</a></li></ul>
<div class="tab-content">
<div id="tabset-4-1">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-58_c1d512be4d078cd93d5238399ff8199c">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-58-1.png" width="1152"></p>
</div>
</div>
</div>
<div id="tabset-4-2">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-59_8a95e61f8720d470f4adf9b100238dae">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-59-1.png" width="1152"></p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="true-vs.-spurious-sub-communities-diagnostic-scores" class="slide level3">
<h3>‚ÄúTrue‚Äù <em>vs.</em> ‚ÄúSpurious‚Äù sub-communities: diagnostic scores</h3>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="images/alto_illustration.png" height="250"></p>
</figure>
</div>
<div class="columns">
<div class="column fragment" style="width:33%;">
<p><em>Plateau</em> in the <strong>number of paths</strong> in the alignment graph at <span class="math inline">\(K^*\)</span>.</p>
<div style="font-size: 60%;">
<p>Path<span class="math inline">\((v)\)</span> of a node <span class="math inline">\(v \in V_m\)</span> is set to Path(<span class="math inline">\(v*\)</span>), where <span class="math inline">\(v^* := \arg \max\limits_{v' \in V_{(m+1):M}} (w_{\text{out}}(v, v') + w_{\text{in}}(v, v'))\)</span></p>
</div>
<p><img data-src="images/path_illustration.png" height="150"></p>
</div><div class="column fragment" style="width:33%;">
<p>Maximum <strong>coherence</strong> scores: how similar is a topic to the other topics on the same path?</p>
<div style="font-size: 60%;">
<p><span class="math inline">\(c(v) = \frac{1}{|\mathcal{P}(v)|} \sum_{v'\in \mathcal{P}(v)} \min\left(w_{\text{out}}(v, v') + w_{\text{in}}(v, v')\right)\)</span></p>
<p>where <span class="math inline">\(\mathcal{P}(v) = \{v': \text{Path}(v') = \text{Path}(v)\}\)</span>.</p>
</div>
<p><img data-src="images/coherence_illustration.png" height="150"></p>
</div><div class="column fragment" style="width:33%;">
<p>Elbow in the <strong>refinement</strong> scores: is this topic a good ‚Äúparent‚Äù topic for the downstream topics?</p>
<div style="font-size: 60%;">
<p><span class="math inline">\(r(v) = \frac{|V_m|}{M - m} \sum_{m' = m+1}^M \sum_{v'_{m'} \in V_{m'}} w_{\text{out}}(v, v'_{m'}) w_{\text{in}}(v, v'_{m'})\)</span></p>
</div>
<p><img data-src="images/refinement_illustration.png" height="150"></p>
</div>
</div>
</section>
<section id="true-vs.-spurious-sub-communities-diagnostic-scores-1" class="slide level3">
<h3>‚ÄúTrue‚Äù <em>vs.</em> ‚ÄúSpurious‚Äù sub-communities: diagnostic scores</h3>
<div class="panel-tabset">
<ul id="tabset-5" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-5-1">Isala (vag. swabs)</a></li><li><a href="#tabset-5-2">LLD (stool)</a></li></ul>
<div class="tab-content">
<div id="tabset-5-1">
<div class="columns">
<div class="column fragment" style="width:33%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-60_aa57472feaf95edf82395ed0c1b455d5">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-60-1.png" width="288"></p>
</div>
</div>
</div><div class="column fragment" style="width:33%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-61_7ac775c77214992eedffdec67a7c5681">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-61-1.png" width="288"></p>
</div>
</div>
</div><div class="column fragment" style="width:33%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-62_88e2de7f009784c3b3b15ec4fbf54a12">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-62-1.png" width="288"></p>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-5-2">
<div class="columns">
<div class="column fragment" style="width:33%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-63_79bd8860e3c04144801c20aa984ed581">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-63-1.png" width="288"></p>
</div>
</div>
</div><div class="column fragment" style="width:33%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-64_551cb37016b43f4a1319419b438b9554">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-64-1.png" width="288"></p>
</div>
</div>
</div><div class="column fragment" style="width:33%;">
<div class="cell" data-hash="LDA-for-microbiota_cache/revealjs/unnamed-chunk-65_9f4a2761f9840689402185128f04e582">
<div class="cell-output-display">
<p><img data-src="LDA-for-microbiota_files/figure-revealjs/unnamed-chunk-65-1.png" width="288"></p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
</section>
<section>
<section id="conclusions-references" class="title-slide slide level1 agenda-slide center" data-background-color="#DDF1F4">
<h1>Conclusions &amp; references</h1>
<div class="agenda-heading">
<p>Outline</p>
</div>
<div class="agenda">
<div class="agenda-inactive agenda-pre-active">
<p>Hello üëãüëã</p>
</div>
<div class="agenda-inactive agenda-pre-active">
<p>Introduction</p>
</div>
<div class="agenda-inactive agenda-pre-active">
<p>Topic models</p>
</div>
<div class="agenda-inactive agenda-pre-active">
<p>Choosing <span class="math inline">\(K\)</span>, the number of topics</p>
</div>
<div class="agenda-active">
<p>Conclusions &amp; references</p>
</div>
</div>
</section>
<section id="summary" class="slide level3">
<h3>Summary</h3>
<ul>
<li>Bacteria interact with each other in complex ways.</li>
</ul>
<div class="fragment">
<ul>
<li>These positive/negative interactions can lead to the existence of <em>subcommunities</em>: groups of bacteria that are more likely to be found together.</li>
</ul>
<p><img data-src="images/interactions_subcommunities.png" width="200"></p>
</div>
<div class="fragment">
<ul>
<li>Topic models (<em>e.g.</em>, LDA) can be used to identify subcommunities from microbiota count data.</li>
</ul>
</div>
<div class="fragment">
<ul>
<li>The number of topics <span class="math inline">\(K\)</span> can be chosen based on (CV) perplexity or through <strong>topic alignment</strong> across <span class="math inline">\(K\)</span>.</li>
</ul>
<p><img data-src="images/alto_illustration.png" width="200"></p>
</div>
<div class="fragment">
<ul>
<li><p>Topic models also serve to reduce the dimensionality of the data.</p>
<ul>
<li><p>Topics can be used as features in downstream analyses.</p></li>
<li><p>For that purpose, it is typically <strong>more powerful than clustering</strong> approaches as clusters do not exist in most microbiotas.</p></li>
</ul></li>
</ul>
</div>
</section>
<section id="alto-package" class="slide level3">
<h3><code>alto</code> package</h3>
<p><a href="https://github.com/lasy/alto"><code>alto</code></a> is an <code>R</code> package that implements the ‚Äútopics alignment‚Äù methods: <a href="https://lasy.github.io/alto/index.html">lasy.github.io/alto</a></p>
<p><img data-src="images/alto_webpage.png" class="nostretch" height="300"></p>
</section>
<section id="references" class="slide level3 smaller scrollable">
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-lda" class="csl-entry" role="listitem">
Blei, David M., Andrew Y. Ng, and Michael I. Jordan. 2003. <span>‚ÄúLatent Dirichlet Allocation.‚Äù</span> <em>J. Mach. Learn. Res.</em> 3 (null): 993‚Äì1022.
</div>
<div id="ref-valencia" class="csl-entry" role="listitem">
France, Michael T., Bing Ma, Pawel Gajer, Sarah Brown, Michael S. Humphrys, Johanna B. Holm, L. Elaine Waetjen, Rebecca M. Brotman, and Jacques Ravel. 2020. <span>‚Äú<span>VALENCIA</span>: A Nearest Centroid Classification Method for Vaginal Microbial Communities Based on Composition.‚Äù</span> <em>Microbiome</em> 8 (1): 166. <a href="https://doi.org/10.1186/s40168-020-00934-6">https://doi.org/10.1186/s40168-020-00934-6</a>.
</div>
<div id="ref-alto" class="csl-entry" role="listitem">
Fukuyama, Julia, Kris Sankaran, and Laura Symul. 2023. <span>‚ÄúMultiscale Analysis of Count Data Through Topic Alignment.‚Äù</span> <em>Biostatistics</em> 24 (4): 1045‚Äì65. <a href="https://doi.org/10.1093/biostatistics/kxac018">https://doi.org/10.1093/biostatistics/kxac018</a>.
</div>
<div id="ref-topicmodels" class="csl-entry" role="listitem">
Gr√ºn, Bettina, and Kurt Hornik. 2011. <span>‚ÄúTopicmodels: An r Package for Fitting Topic Models.‚Äù</span> <em>Journal of Statistical Software</em>.
</div>
<div id="ref-isala" class="csl-entry" role="listitem">
Lebeer, Sarah, Sarah Ahannach, Thies Gehrmann, Stijn Wittouck, Tom Eilers, Eline Oerlemans, Sandra Condori, et al. 2023. <span>‚ÄúA Citizen-Science-Enabled Catalogue of the Vaginal Microbiome and Associated Factors.‚Äù</span> <em>Nature Microbiology</em> 8 (11): 2183‚Äì95. <a href="https://doi.org/10.1038/s41564-023-01500-0">https://doi.org/10.1038/s41564-023-01500-0</a>.
</div>
<div id="ref-curatedMg" class="csl-entry" role="listitem">
Pasolli, Edoardo, Lucas Schiffer, Paolo Manghi, Audrey Renson, Valerie Obenchain, Duy Tin Truong, Francesco Beghini, et al. 2017. <span>‚ÄúAccessible, Curated Metagenomic Data Through <span>ExperimentHub</span>.‚Äù</span> <em>Nature Methods</em> 14 (11): 1023‚Äì24. <a href="https://doi.org/10.1038/nmeth.4468">https://doi.org/10.1038/nmeth.4468</a>.
</div>
<div id="ref-lda1" class="csl-entry" role="listitem">
Pritchard, JK, M Stephens, and P Donnelly. 2000. <span>‚ÄúInferring Population Structure Using Multilocus Genotype Data.‚Äù</span> <em>Genetics</em>.
</div>
<div id="ref-sankaran_latent_2019" class="csl-entry" role="listitem">
Sankaran, Kris, and Susan P Holmes. 2019. <span>‚ÄúLatent Variable Modeling for the Microbiome.‚Äù</span> <em>Biostatistics</em> 20 (4): 599‚Äì614. <a href="https://doi.org/10.1093/biostatistics/kxy018">https://doi.org/10.1093/biostatistics/kxy018</a>.
</div>
<div id="ref-subcommunities" class="csl-entry" role="listitem">
Symul, Laura, Pratheepa Jeganathan, Elizabeth K. Costello, Michael France, Seth M. Bloom, Douglas S. Kwon, Jacques Ravel, David A. Relman, and Susan Holmes. 2023. <span>‚ÄúSub-Communities of the Vaginal Microbiota in Pregnant and Non-Pregnant Women.‚Äù</span> <em>Proceedings of the Royal Society B: Biological Sciences</em> 290 (2011): 20231461. <a href="https://doi.org/10.1098/rspb.2023.1461">https://doi.org/10.1098/rspb.2023.1461</a>.
</div>
</div>
</section>
<section id="thank-you" class="slide level3" data-background-image="images/cover.png" data-background-size="cover" data-background-opacity="0.5">
<h3>Thank you!</h3>
<p>Questions?</p>
<p><img src="https://avatars.githubusercontent.com/u/7518113?v=4" class="slide-logo"></p>
<div class="footer footer-default">
<p><a href="https://github.com/lasy/t-2425-Microbiota-UHasselt">github.com/lasy/t-2425-Microbiota-UHasselt</a></p>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="LDA-for-microbiota_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/multiplex/socket.io.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/multiplex/multiplex.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="LDA-for-microbiota_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":false},
'chalkboard': {"buttons":true,"boardmarkerWidth":2,"theme":"chalkboard","chalkWidth":3},
'multiplex': {"secret":null,"id":"dc0367525792c7d6","url":"https://reveal-multiplex.glitch.me/"},
'smaller': true,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: true,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>